src/PointwiseFunctions/Elasticity/ConstitutiveRelations/CubicCrystal.cpp

```diff
@@ -0,0 +1,83 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/CubicCrystal.hpp"
+
+#include <array>
+#include <pup.h>  // IWYU pragma: keep
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace Elasticity::ConstitutiveRelations {
+
+CubicCrystal::CubicCrystal(double c_11, double c_12, double c_44) noexcept
+    : c_11_(c_11), c_12_(c_12), c_44_(c_44) {
+  ASSERT(
+      c_11_ >= c_12_,
+      "c_11 must be bigger than c_12, but are c_11="
+          << c_11 << " and c_12=" << c_12
+          << ". This is because the youngs_modulus "
+             "must be positive and the poisson ratio smaller or equal to 0.5.");
+}
+
+tnsr::II<DataVector, 3> CubicCrystal::stress(
+    const tnsr::ii<DataVector, 3>& strain,
+    const tnsr::I<DataVector, 3>& /*x*/) const noexcept {
+  auto result = make_with_value<tnsr::II<DataVector, 3>>(strain, 0.);
+  for (size_t i = 0; i < 3; i++) {
+    for (size_t j = 0; j <= i; j++) {
+      result.get(i, j) = -2. * c_44_ * strain.get(i, j);
+    }
+  }
+  auto trace = make_with_value<DataVector>(strain, 0.);
+  for (size_t i = 0; i < 3; i++) {
+    trace += strain.get(i, i);
+  }
+  for (size_t i = 0; i < 3; i++) {
+    result.get(i, i) -=
```

Comment:
move this to the first loop (removing the c_44 term as it is only on the off-diagonal and replacing trace with result.get(2,2))